torch/optim/adamax.py

```diff
@@ -331,12 +341,14 @@ def _multi_tensor_adamax(
         torch._foreach_mul_(grouped_exp_infs, beta2)
 
         for exp_inf, grad in zip(grouped_exp_infs, grouped_grads):
-            norm_buf = torch.cat(
-                [exp_inf.unsqueeze(0), grad.abs().add_(eps).unsqueeze_(0)], 0
+            torch.maximum(
```

Comment:
@janeyx99 we were discussing this before I left on PTO.is this a correct impl? @jon-chuang you should likely use the foreach ops here instead of a loop if possible.

Reply:
I don't think there is a foreach_max nor foreach_maximum

Reply:
@mlazos if you want, I can create a separate PR implementing foreach max/maximum?

Reply:
Reference on steps necessary to add a new foreach operator:
https://github.com/pytorch/pytorch/pull/106574

Reply:
There is definitely a foreach maximum

Reply:
hmm, I guess it is not in the docs: https://pytorch.org/docs/stable/search.html?q=foreach&check_keywords=yes&area=default#

Maybe it should be? Or maybe its fine as its an internal API